src/detours.cpp

```diff
@@ -180,8 +182,24 @@ void FASTCALL Detour_Host_Say(CCSPlayerController *pController, CCommand &args,
 	bool bGagged = pController && g_playerManager->GetPlayer(pController->GetPlayerSlot())->IsGagged();
 
 	if (!bGagged && *args[1] != '/')
+	{
 		Host_Say(pController, args, teamonly, unk1, unk2);
 
+		if (pController)
+		{
+			IGameEvent *pEvent = g_gameEventManager->CreateEvent("player_chat");
+
+			if (pEvent)
+			{
+				pEvent->SetBool("teamonly", teamonly);
+				pEvent->SetInt("userid", pController->GetPlayerSlot());
```

Comment:
This should probably use entindex rather than the slot. Slots are just the entity index - 1.